### ISSUE TITLE
Support new UI5 "Modules"

### ISSUE THREAD

**Issue by bd82** [SAP/ui5-typescript#11]
We need to investigate how to model this in a d.ts file.
- As a TypeScript module it may not mean anything as UI5 have their own model system,
- As a global namespace it may incorrectly expose none global APIs as modules are only usable via sap.ui.define.

Perhaps expose as a specially named namespace?
```typescript
declare namespace modules {
   namespace sap {
      namespace base {
       // ....
       } 
   }
}


sap.ui.define(/* ...  */, 
/**
 * @param {modules.sap.base.bamba.A} A
/* 
function(A,b,c) {

})
```

**Comment by bd82** [comment]
Maybe "import types" would be relevant?
![alt text](https://cdn-images-1.medium.com/max/2000/0*U2IODcFsEn9llZT7.gif)

**Comment by codeworrior** [comment]
I rather would model them as real typescript modules:

```ts
declare module "sap/base/i18n/ResourceBundle" {
    export default class ResourceBundle {
        getText(key:string): string
        static create() : ResourceBundle
    }
}
```

This allows to refer to them in signatures 

```ts
declare namespace sap {
  ...
    class ResourceModel extends sap.ui.model.Model {
      constructor( 
          bundle? : typeof import("sap/base/i18n/ResourceBundle") | any
      )
}
```

and it preserves the scoping of modules (no global names, no naming conflicts with other names).

**Comment by nlunets** [comment]
Technically why would you even need the namespace in a modern UI5 development ?
People don't usually refers to the namespace value in modern ui5 (expect for getting the sap.ui.getCore()) so exposing a namespace may lead to bad practice compared to using module definition :)

**Comment by bd82** [comment]
Yes, we could possibly avoid exporting global namespaces and only export modules
maybe we do not need to support super legacy global style UI5 code and only support `sap.ui.define` style or other module systems

**Comment by petermuessig** [comment]
The generator will get support for the generation of the `ts-types-esm` with the upcoming `3.0` release.

**Comment by akudev** [comment]
...so this is long done.
